engine/src/main.cpp

```diff
@@ -306,7 +306,9 @@ int main( int argc, char *argv[] )
     std::string program_path_str(argv[0]);
     boost::filesystem::path program_path(program_path_str);
     boost::filesystem::path program_directory_path = program_path.parent_path();
-    boost::filesystem::current_path(program_directory_path);
+    if ( ! program_directory_path.empty())                  // Changing to an empty path does bad things
+    {    boost::filesystem::current_path(program_directory_path);
+    }
```

Comment:
Probably should generate a warning if it doesn't change the path.

Also, not sure why GitHub is formatting the code line to be on the same line as the first brace...
Should be:
```
if (...)
{
    ...
}
```
Please verify

Reply:
Ah, it was not Github's fault. 
In my coding studies I have picked up the habit of putting the first statement after an opening brace on the same line as the brace, rather than on it's own line. So, completely my fault, and is corrected now.

Reply:
@evertvorster fully understand, my personal preferred C++ style is:
```
block statement
   {
   // block code
   }
```

I find that much more readable than putting the braces elsewhere; it seems to be unique enough it's very hard to get code generators/formaters to do that style; so I've had to learn to adapt to what others use.